lib/resty/openssl/x509/csr.lua

```diff
@@ -11,14 +9,18 @@ local stack_lib = require "resty.openssl.stack"
 local pkey_lib = require "resty.openssl.pkey"
 local altname_lib = require "resty.openssl.x509.altname"
 local digest_lib = require("resty.openssl.digest")
+local extension_lib = require("resty.openssl.x509.extension")
 local util = require "resty.openssl.util"
+local txtnid2nid = require("resty.openssl.objects").txtnid2nid
 local format_error = require("resty.openssl.err").format_error
 local OPENSSL_10 = require("resty.openssl.version").OPENSSL_10
 local OPENSSL_11_OR_LATER = require("resty.openssl.version").OPENSSL_11_OR_LATER
-
+local ext_typ_ptr = "X509_EXTENSION" .. "*"
 local accessors = {}
-
-
+local C = ffi.C
+local ffi_gc = ffi.gc
+local ffi_cast = ffi.cast
+local push = table.insert
```

Comment:
let's move this block at top after `local ffi = require "ffi"`

Reply:
I prefer to have all require block on top, than locals, and module declaration after it. What do you think?

Reply:
It's usually a pattern for those "cached" imports at top and separate from other normal locals, we can take https://github.com/openresty/lua-resty-core/blob/master/lib/resty/core/ctx.lua as an exmaple

Reply:
ok